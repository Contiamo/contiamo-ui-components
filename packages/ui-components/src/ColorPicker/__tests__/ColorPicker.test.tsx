import * as React from "react"
import { shallow, mount } from "enzyme"
import { jsdom } from "jsdom"

import wrapDefaultTheme from "../../../utils/wrap-default-theme"
import ThemelessColorPicker from "../ColorPicker"

const ColorPicker = wrapDefaultTheme(ThemelessColorPicker)

/*
  We need a DOM in order to test:
  - Click outside to close
  - Press escape to close
*/
global.document = jsdom('<!doctype html><html><body><div id="root"></div></body></html>')
global.window = document.parentWindow

describe("ColorPicker", () => {
  let comp
  beforeEach(() => {
    comp = mount(<ColorPicker />, {
      attachTo: global.document.getElementById("root")
    })
  })
  it("Should initialize properly", () => {
    expect(comp).toMatchSnapshot()
  })
  it("Should be toggled on click", () => {
    comp.simulate("click")
    expect(comp.state().isPickerOpen).toBe(true)

    comp.simulate("click")
    expect(comp.state().isPickerOpen).toBe(false)
  })

  it("Should close on press of Esc", () => {
    const pressEsc = new KeyboardEvent("keyup", { keyCode: 27 })

    comp.simulate("click")
    expect(comp.state().isPickerOpen).toBe(true)

    document.dispatchEvent(pressEsc)
    expect(comp.state().isPickerOpen).toBe(false)
  })

  it("Should close on click outside", () => {
    const clickOutside = new MouseEvent("click")

    comp.simulate("click")
    expect(comp.state().isPickerOpen).toBe(true)

    document.dispatchEvent(clickOutside)
    expect(comp.state().isPickerOpen).toBe(false)
  })

  it("Should gracefully unmount", () => {
    expect(comp.unmount().component.state.mount).toBe(false)
  })
})
