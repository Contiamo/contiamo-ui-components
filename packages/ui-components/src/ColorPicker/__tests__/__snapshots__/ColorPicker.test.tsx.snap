// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Should initialize properly 1`] = `
<ColorPicker
  color="#03f"
  size={16}
>
  <div
    onClick={[Function]}
  >
    <glamorous(div)
      color="#03f"
      size={16}
    >
      <div
        className="css-ebsmcl"
      />
    </glamorous(div)>
  </div>
</ColorPicker>
`;
